### PR TITLE
[ADD] project_budget_management: Added project budget management module

### PR DESCRIPTION
**Pull request by Trivedi-Vacha-SerpentCS**

### CODE REVIEW

File: project_budget_management/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+{
+    "name": "Project Budget Management",
+    "version": "15.0.1.0.0",
+    "summary": "A Module adds features to the Project Budget Management.",
+    "author": """Serpent Consulting Services Pvt. Ltd.""",
+    "website": "http://www.serpentcs.com",
+    "category": "Project Budget Management",
+    "license": "LGPL-3",
+    "depends": ["project_timesheet_management"],
```
Comment (bvanjeelharia) [inline project_budget_management/__manifest__.py:11]: Where is this module?

File: project_budget_management/models/project_project.py [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+                inv.total_amount for inv in timesheet_invoice_records
+            )
+
+    @api.depends("actual_budget", "revised_budget")
+    def _compute_actual_revised_budget(self):
+        for project in self:
+            actual_revised_budget = project.actual_budget
+            if project.revised_budget > 0:
+                actual_revised_budget = project.revised_budget
+            project.actual_revised_budget = actual_revised_budget
```
Comment (bvanjeelharia) [inline project_budget_management/models/project_project.py]: Simplify this as 
project.actual_revised_budget = project.revised_budget if project.revised_budget > 0 else project.actual_budget

File: project_budget_management/models/project_project.py [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+                        (100 - project.percentage_completed)
+                        / project.percentage_completed
+                    )
+                    * project.spent_budget
+                ) + project.spent_budget
+            project.budget_of_completion = budget_of_completion
+
+    def _compute_assignee_id_editable(self):
+        cann_pro_crd = self.env.user.has_group(
+            "project_management_security.group_project_coordinator"
```
Comment (bvanjeelharia) [inline project_budget_management/models/project_project.py:74]: dependency for project_management_security module is missing in manifest

File: project_budget_management/models/project_project.py [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+            ) else False
+
+    @api.depends('date_start', 'percentage_completed', 'expected_end_date', 'stage_id')
+    def _compute_projected_end_date(self):
+        for project in self:
+            proj_end_dt = False
+            if (project.percentage_completed > 0 and
+                    project.date_start and
+                    project.stage_id.is_close is not True):
+                today_date = fields.Datetime.today().date()
```
Comment (bvanjeelharia) [inline project_budget_management/models/project_project.py]: Move it outside of for loop

File: project_budget_management/models/project_project.py [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+                start_date = project.date_start
+                spent_days = 0
+                if start_date:
+                    spent_days = (today_date - start_date).days
```
Comment (bvanjeelharia) [inline project_budget_management/models/project_project.py]: Change this to spent_days = (today_date - start_date).days if start_date else 0

File: project_budget_management/wizard/project_approve_wiz.py [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+                        rev = "Budget Revisions"
+                    elif self._context.get("extend_field") == "expected_end_date":
+                        parent_rec.project_id.write(
+                            {
+                                "expected_end_date": parent_rec.new_expected_end_date,
+                                "rev_date": fields.Datetime.now(),
+                            }
+                        )
+                        rev = "Project Revisions"
+                msg = (
```
Comment (bvanjeelharia) [inline project_budget_management/wizard/project_approve_wiz.py:45]: Shouldn't the msg be posted if only there's a value for rev?

**Review by bvanjeelharia — CHANGES_REQUESTED** [review]

File: project_budget_management/__manifest__.py (hunk 1 above)
Comment (Trivedi-Vacha-SerpentCS) [reply project_budget_management/__manifest__.py:11]: Will add pr of timesheet module today, but kindly note that it still require code level improvements as its WIP.

Comment (bvanjeelharia) [reply project_budget_management/__manifest__.py:11]: ok make sure all the base modules are added before add other modules

File: project_budget_management/models/project_project.py (hunk 3 above)
Comment (Trivedi-Vacha-SerpentCS) [reply project_budget_management/models/project_project.py:74]: module is indirectly depends on project_management_security

File: project_budget_management/wizard/project_approve_wiz.py (hunk 6 above)
Comment (Trivedi-Vacha-SerpentCS) [reply project_budget_management/wizard/project_approve_wiz.py:45]: Tested and compared with old version, 
It is working as expected, no change require

**Comment by Trivedi-Vacha-SerpentCS** [comment]
@bvanjeelharia Kindly verify

**Pull request merged**